langchain4j/src/main/java/dev/langchain4j/data/document/loader/S3DirectoryLoader.java

```diff
@@ -0,0 +1,124 @@
+package dev.langchain4j.data.document.loader;
+
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.DocumentType;
+import dev.langchain4j.data.document.source.S3Source;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static dev.langchain4j.data.document.loader.DocumentLoaderUtils.parserFor;
+import static dev.langchain4j.internal.Utils.isNullOrBlank;
+
+/**
+ * S3 Directory Loader Implementation
+ */
+public class S3DirectoryLoader extends AbstractS3Loader<List<Document>> {
+
+    private static final Logger log = LoggerFactory.getLogger(S3DirectoryLoader.class);
+
+    private final String prefix;
+
+    private S3DirectoryLoader(Builder builder) {
+        super(builder);
+
+        if (isNullOrBlank(bucket)) {
+            throw new IllegalArgumentException("Bucket is a required parameter.");
```

Comment:
i guess this check should be in the parent constructor

Reply:
removed this line and added necessary validations at constructor